be/src/vec/functions/array/function_array_distinct.h

```diff
@@ -0,0 +1,297 @@
+                        size_t result, size_t input_rows_count) override {
+        ColumnPtr src_column =
+                block.get_by_position(arguments[0]).column->convert_to_full_column_if_const();
+        const auto& src_column_array = check_and_get_column<ColumnArray>(*src_column);
+        if (!src_column_array) {
+            return Status::RuntimeError(
+                    fmt::format("unsupported types for function {}({})", get_name(),
+                                block.get_by_position(arguments[0]).type->get_name()));
+        }
+        const auto& src_offsets = src_column_array->get_offsets();
+        const auto& src_nested_column = src_column_array->get_data();
+
+        DataTypePtr src_column_type = remove_nullable(block.get_by_position(arguments[0]).type);
+        auto nested_type = assert_cast<const DataTypeArray&>(*src_column_type).get_nested_type();
+        auto dest_column_ptr = ColumnArray::create(nested_type->create_column(),
+                                                   ColumnArray::ColumnOffsets::create());
+        IColumn& dest_nested_column = dest_column_ptr->get_data();
+        ColumnArray::Offsets& dest_offsets = dest_column_ptr->get_offsets();
+        dest_nested_column.reserve(src_nested_column.size());
+        dest_offsets.reserve(input_rows_count);
+
+        const IColumn* src_inner_column = nullptr;
+        const ColumnNullable* src_nested_nullable_col = nullptr;
+        if (src_nested_column.is_nullable()) {
+            src_nested_nullable_col = check_and_get_column<ColumnNullable>(src_nested_column);
+            src_inner_column = src_nested_nullable_col->get_nested_column_ptr();
+        } else {
+            src_inner_column = src_column_array->get_data_ptr();
+        }
+
+        IColumn* dest_inner_column = nullptr;
+        ColumnNullable* dest_nested_nullable_col = nullptr;
+        if (dest_nested_column.is_nullable()) {
+            dest_nested_nullable_col = reinterpret_cast<ColumnNullable*>(&dest_nested_column);
+            dest_inner_column = &dest_nested_nullable_col->get_nested_column();
+        } else {
+            dest_inner_column = &dest_nested_column;
+        }
+
+        auto res_val =
+                _execute_by_type(*src_inner_column, src_offsets, *dest_inner_column, dest_offsets,
+                                 src_nested_nullable_col, dest_nested_nullable_col, nested_type);
+        if (!res_val) {
+            return Status::RuntimeError(
+                    fmt::format("execute failed or unsupported types for function {}({})",
+                                get_name(), block.get_by_position(arguments[0]).type->get_name()));
+        }
+
+        block.replace_by_position(result, std::move(dest_column_ptr));
+        return Status::OK();
+    }
+
+private:
+    // Note: Here initially allocate a piece of memory for 2^5 = 32 elements.
+    static constexpr size_t INITIAL_SIZE_DEGREE = 5;
+
+    template <typename ColumnType, typename NestType>
```

Comment:
ColumnType or NestType, only one is needed:
Input ColumnType, using NestType = ColumnType::value_type
Input NestType, using ColumnType = ColumnVector<NestType>